tests/unit/stream_alert_rule_processor/test_rules_engine.py

```diff
@@ -490,8 +490,11 @@ def match_ipaddress(rec): # pylint: disable=unused-variable
               outputs=['s3:sample_bucket'],
               datatypes=['sourceAddress', 'command'])
         def mismatch_types(rec): # pylint: disable=unused-variable
-            """Testing rule with non-existing normalized type in the record. It
-            should not trigger alert.
+            """Testing rule with non-existing normalized type in the record.
```

Comment:
This description is really confusing :(

What are you testing?

1) a rule has 2 datatypes defined, but only 1 exists in the record?
2) a rule has 2 datatypes defined, but none exist in the record?
3) a rule has 2 datatypes defined, but both exist in the record?

With the existing description, it's not clear why we'd see a "trigger alert" if there's a record coming into a rule that does not have the specified datatype